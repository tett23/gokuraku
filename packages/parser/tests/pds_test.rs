extern crate parser;

use insta::{assert_debug_snapshot, glob};
use std::fs;

#[test]
fn pds_test() {
    glob!("../fixtures/pds", "*.pds", |path| {
        let input = fs::read_to_string(path).unwrap();
        let ast = parser::prose_down_script_parse(&input).unwrap();

        assert_debug_snapshot!(ast);
    });
}
