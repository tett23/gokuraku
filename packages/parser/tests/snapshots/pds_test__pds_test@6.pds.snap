---
source: packages/parser/tests/pds_test.rs
expression: ast
input_file: packages/parser/fixtures/pds/6.pds
---
Module {
    statements: [
        DataAssign(
            DataAssign {
                modifier: None,
                constraints: [],
                ident: ExistsIdent(
                    ExistsIdent(
                        "Bool",
                    ),
                ),
                args: [],
                expr: Or(
                    Value(
                        Constructor(
                            DataConstructor {
                                modifier: None,
                                ident: ExistsIdent(
                                    ExistsIdent(
                                        "True",
                                    ),
                                ),
                                args: [],
                            },
                        ),
                    ),
                    Value(
                        Constructor(
                            DataConstructor {
                                modifier: None,
                                ident: ExistsIdent(
                                    ExistsIdent(
                                        "False",
                                    ),
                                ),
                                args: [],
                            },
                        ),
                    ),
                ),
            },
        ),
        DataAssign(
            DataAssign {
                modifier: Some(
                    Nominal,
                ),
                constraints: [],
                ident: ExistsIdent(
                    ExistsIdent(
                        "True",
                    ),
                ),
                args: [],
                expr: Value(
                    Unit,
                ),
            },
        ),
        DataAssign(
            DataAssign {
                modifier: Some(
                    Nominal,
                ),
                constraints: [],
                ident: ExistsIdent(
                    ExistsIdent(
                        "False",
                    ),
                ),
                args: [],
                expr: Value(
                    Unit,
                ),
            },
        ),
        DataAssign(
            DataAssign {
                modifier: Some(
                    Nominal,
                ),
                constraints: [],
                ident: ExistsIdent(
                    ExistsIdent(
                        "Int",
                    ),
                ),
                args: [],
                expr: Value(
                    Unit,
                ),
            },
        ),
        AssignDef(
            AssignDef {
                ident: Ident(
                    "id",
                ),
                expr: TypeExpr {
                    trait_constraints: [],
                    eta_envs: EtaEnvs(
                        [],
                    ),
                    expr: Literal(
                        Constructor(
                            DataConstructor {
                                modifier: None,
                                ident: ExistsIdent(
                                    ExistsIdent(
                                        "Int",
                                    ),
                                ),
                                args: [],
                            },
                        ),
                    ),
                },
            },
        ),
        AssignDef(
            AssignDef {
                ident: Ident(
                    "unary",
                ),
                expr: TypeExpr {
                    trait_constraints: [],
                    eta_envs: EtaEnvs(
                        [],
                    ),
                    expr: Arrow(
                        Literal(
                            Constructor(
                                DataConstructor {
                                    modifier: None,
                                    ident: ExistsIdent(
                                        ExistsIdent(
                                            "Int",
                                        ),
                                    ),
                                    args: [],
                                },
                            ),
                        ),
                        Literal(
                            Constructor(
                                DataConstructor {
                                    modifier: None,
                                    ident: ExistsIdent(
                                        ExistsIdent(
                                            "Int",
                                        ),
                                    ),
                                    args: [],
                                },
                            ),
                        ),
                    ),
                },
            },
        ),
        AssignDef(
            AssignDef {
                ident: Ident(
                    "binary",
                ),
                expr: TypeExpr {
                    trait_constraints: [],
                    eta_envs: EtaEnvs(
                        [],
                    ),
                    expr: Arrow(
                        Literal(
                            Constructor(
                                DataConstructor {
                                    modifier: None,
                                    ident: ExistsIdent(
                                        ExistsIdent(
                                            "Int",
                                        ),
                                    ),
                                    args: [],
                                },
                            ),
                        ),
                        Arrow(
                            Literal(
                                Constructor(
                                    DataConstructor {
                                        modifier: None,
                                        ident: ExistsIdent(
                                            ExistsIdent(
                                                "Int",
                                            ),
                                        ),
                                        args: [],
                                    },
                                ),
                            ),
                            Literal(
                                Constructor(
                                    DataConstructor {
                                        modifier: None,
                                        ident: ExistsIdent(
                                            ExistsIdent(
                                                "Int",
                                            ),
                                        ),
                                        args: [],
                                    },
                                ),
                            ),
                        ),
                    ),
                },
            },
        ),
        AssignDef(
            AssignDef {
                ident: Ident(
                    "ternary",
                ),
                expr: TypeExpr {
                    trait_constraints: [],
                    eta_envs: EtaEnvs(
                        [],
                    ),
                    expr: Arrow(
                        Literal(
                            Constructor(
                                DataConstructor {
                                    modifier: None,
                                    ident: ExistsIdent(
                                        ExistsIdent(
                                            "Int",
                                        ),
                                    ),
                                    args: [],
                                },
                            ),
                        ),
                        Arrow(
                            Literal(
                                Constructor(
                                    DataConstructor {
                                        modifier: None,
                                        ident: ExistsIdent(
                                            ExistsIdent(
                                                "Int",
                                            ),
                                        ),
                                        args: [],
                                    },
                                ),
                            ),
                            Arrow(
                                Literal(
                                    Constructor(
                                        DataConstructor {
                                            modifier: None,
                                            ident: ExistsIdent(
                                                ExistsIdent(
                                                    "Int",
                                                ),
                                            ),
                                            args: [],
                                        },
                                    ),
                                ),
                                Literal(
                                    Constructor(
                                        DataConstructor {
                                            modifier: None,
                                            ident: ExistsIdent(
                                                ExistsIdent(
                                                    "Int",
                                                ),
                                            ),
                                            args: [],
                                        },
                                    ),
                                ),
                            ),
                        ),
                    ),
                },
            },
        ),
    ],
}
